### PR TITLE
E2E tests workflow

### PR DESCRIPTION
**Pull request by codemonkey800**
#424

- Creates reusuable workflow for running E2E tests
- Adds workflow after staging deployments
- Fixes issue with E2E tests being flaky on initial load

### CODE REVIEW

File: .github/workflows/staging-deploy.yml [hunk 1]
```diff
@@ -3,7 +3,9 @@ name: Deploy to staging
 on:
   push:
     branches:
-      - "main"
+      - main
+    paths:
+      - 'frontend/**'
```
Comment (manasaV3) [inline .github/workflows/staging-deploy.yml:8]: We should also add happy path here.

File: .github/workflows/frontend-e2e-tests.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
+on:
+  workflow_call:
+    inputs:
```
Comment (manasaV3) [inline .github/workflows/frontend-e2e-tests.yml:3]: Reusable workflow components for the win! 🎉

File: .github/workflows/frontend-e2e-tests.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    secrets:
+      E2E_CONFIG:
+        required: true
+
+jobs:
+  frontend-e2e-tests:
+    strategy:
+      matrix:
+        browser: [chrome]
```
Comment (manasaV3) [inline .github/workflows/frontend-e2e-tests.yml:16]: Curious about why we only test for Chrome?

File: .github/workflows/staging-deploy.yml (hunk 1 above)
Comment (manasaV3) [reply .github/workflows/staging-deploy.yml:8]: Forgot, we moved the `.happy` folder inside `frontend`.

File: frontend/packages/data-portal/e2e/filters/utils.ts [hunk 4]
```diff
@@ -64,3 +64,19 @@ export async function validateTable({
...
   await validateRows(page)
 }
+
+/**
+ * When loading the page, we need to wait a bit after so that the SDS components
+ * have time to become interactive. Without the timeout, the tests become more
+ * flaky and occasionally fail. For example, the filter dropdowns sometimes
+ * don't open when clicked on because the playwright browser starts clicking on
+ * it too fast while the JavaScript is still loading and hydrating.
+ */
```
Comment (kne42) [inline frontend/packages/data-portal/e2e/filters/utils.ts:74]: weird. how much slower does this make our tests?

**Review by kne42 — APPROVED** [review]

File: .github/workflows/staging-deploy.yml [hunk 5]
```diff
@@ -3,7 +3,10 @@ name: Deploy to staging
...
+      - 'frontend/.happy/**'
```
Comment (manasaV3) [inline .github/workflows/staging-deploy.yml:9]: Sorry, I marked my earlier comment as resolved. We don't need this, as it should be covered by the `frontend/**` line.

**Review by manasaV3 — APPROVED** [review]
🚀

**Pull request merged**